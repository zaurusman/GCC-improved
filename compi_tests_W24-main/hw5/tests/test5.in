print("TRUE evaluation");
if (true)
    printi(1);
else
    printi(0);

print("FALSE evaluation");
if (false)
    printi(1);
else
    printi(0);

print("Evaluation with if (exp){printi(1)}else{printi(0)} with truth tabels {0, 1} and {00, 01, 10, 11}");

print("NOT truth table");
if (not false)
    printi(1);
else
    printi(0);

if (not true)
    printi(1);
else
    printi(0);


print("AND truth table");

if (false and false)
    printi(1);
else
    printi(0);

if (false and true)
    printi(1);
else
    printi(0);

if (true and false)
    printi(1);
else
    printi(0);

if (true and true)
    printi(1);
else
    printi(0);


print("OR truth table");

if (false or false)
    printi(1);
else
    printi(0);

if (false or true)
    printi(1);
else
    printi(0);

if (true or false)
    printi(1);
else
    printi(0);

if (true or true)
    printi(1);
else
    printi(0);


