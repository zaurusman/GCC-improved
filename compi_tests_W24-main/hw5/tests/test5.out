TRUE evaluation
1
FALSE evaluation
0
Evaluation with if (exp){printi(1)}else{printi(0)} with truth tabels {0, 1} and {00, 01, 10, 11}
NOT truth table
1
0
AND truth table
0
0
0
1
OR truth table
0
1
1
1
